hawkular-alerts-actions-api/src/main/java/org/hawkular/alerts/actions/api/PluginOpMessage.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.actions.api;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A message sent to the alerts engine from the plugin
+ * It defines a code of operation and payloads
+ *
+ * It used for plugin registration but additional operations should be supported
+ *
+ * @author Lucas Ponce
+ */
+public interface PluginOpMessage {
```

Comment:
I prefer PluginOperationMessage. PluginMessageType is not meaningful here - I think - as this is an operation message sent by the plugin.
The type is an internal field to define the operation.